Use × instead of x when printing array dimensions, continued

nalimilan opened the pull request:
Commit bde452d (https://github.com/JuliaLang/julia/pull/15910) missed sparse matrices and docs. Also make sparse
array summary line closer to sparse matrices and other arrays.
By the way, fix very old docs in which the summary line for
standard arrays was out of date.

The new output for sparse vectors is:

``` julia
julia> sparse([1,2])
2-element sparse vector with 2 Int64 entries:
  [1]  =  1
  [2]  =  2
```

Compare with sparse matrix and standard vector:

``` julia
julia> sparse([1 2])
1×2 sparse matrix with 2 Int64 entries:
    [1, 1]  =  1
    [1, 2]  =  2

julia> [1,2]
2-element Array{Int64,1}:
 1
 2
```

We could imagine making the printing of sparse matrix/vector close to other arrays, e.g. `1×2 SparseMatrixCSC{Int64} with 2 entries:`.

tkelman commented on base/sparse/sparsevector.jl:
```diff
@@ -636,8 +636,8 @@ function showarray(io::IO, x::AbstractSparseVector;
 function summary(x::AbstractSparseVector)
-    string("Sparse vector of length ", length(x), ", with ", length(nonzeros(x)),
-           " ",  eltype(x), " nonzero entries:")
+    string(length(x), "-element sparse vector with ", length(nonzeros(x)),
```
the meaning of "-element" is a bit non obvious for the sparse case

nalimilan replied:
Yeah, maybe. OTOH, the fact that the number of `"entries"` is printed on the same line removes any ambiguity.

I could also reintroduce the term "nonzero entries", which isn't currently used for matrices, but which could be added for both matrices and vectors.

nalimilan replied:
I've added "nonzero entries" to both matrix and vector. (I'd prefer we say "stored entries", but this would have to be changed in the whole code base, including renaming functions.)

stevengj commented:
See also [the FFT code here](https://github.com/JuliaLang/julia/blob/d6df2a0b166d360b24f4cc3631a65e3cc5c56476/base/fft/FFTW.jl#L270).

nalimilan commented:
@stevengj Thanks. Interestingly, no example of this output appears in the code, which is why I missed it. Anyway, I've added it, but please check that the second change I made is also correct.

stevengj commented on base/fft/FFTW.jl:
```diff
@@ -310,7 +310,7 @@ function show{T,K,inplace}(io::IO, p::r2rFFTWPlan{T,K,inplace})
         end
     else
-        print(io, join(map(kind2string, K), "x"))
+        print(io, join(map(kind2string, K), "×"))
```
looks good

nalimilan commented:
@tkelman I'll merge tomorrow unless you think your objection still stands.

tkelman commented:
I still think element is a bit unclear. There should be a better way of expressing the dimensionality of the vector.

nalimilan commented:
@tkelman Any suggestions then? I can think of "Length 10 sparse vector with 2 Int64 entries:", but this still has the drawback of moving the interesting information (10) to the middle of the string (which would be a regression IMHO if we apply this to `Vector`). Or I can keep the current version, and forget about being consistent with `Vector`.

tkelman commented:
It's a different data structure, I don't mind displaying it a little differently.

nalimilan committed:
Use × instead of x when printing array dimensions, continued

Commit bde452d missed sparse matrices, FFTW and docs. Also make sparse
matrix vector summary line closer to that for vectors.
By the way, fix very old docs in which the summary line for standard
arrays was out of date.

nalimilan commented:
OK, I've kept the old summary for sparse vectors. I've just added the mention of "nonzero entries" to the summary for sparse matrices.

nalimilan commented:
The Travis build was killed without any details during precompile on Linux 64 bit, and timed out on Mac. I've restarted it.

nalimilan commented:
OS X build timed out once again. Other than that it should be good to go.
